File: demo/vb/ConfigFileTests.vb
Imports System.Configuration
Imports System.IO
Imports NUnit.Framework

Namespace NUnitTestDemo

    <ExpectPass>
    Public Class ConfigFileTests
        <Test>
        Public Shared Sub ProperConfigFileIsUsed()
            Dim expectedPath = Path.Combine(TestContext.CurrentContext.TestDirectory, "NUnit3TestDemo.dll.config")
            Assert.That(expectedPath, Iz.EqualTo(AppDomain.CurrentDomain.SetupInformation.ConfigurationFile))
        End Sub

        <Test>
        Public Shared Sub CanReadConfigFile()
            Assert.That(ConfigurationManager.AppSettings.Get("test.setting"), Iz.EqualTo("54321"))
        End Sub

    End Class

End Namespace

File: demo/cpp/ConfigFileTests.cpp
#include "ExpectedOutcomeAttributes.h"

using namespace System;
using namespace System::Configuration;
using namespace System::IO;
using namespace NUnit::Framework;

namespace NUnitTestDemo
{
    [ExpectPass]
    public class ConfigFileTests
    {
	public:
        [Test]
        static void ProperConfigFileIsUsed()
        {
            String^ expectedPath = Path::Combine(TestContext::CurrentContext->TestDirectory, "NUnit3TestDemo.dll.config");
            Assert::That(expectedPath, Is::EqualTo(AppDomain::CurrentDomain->SetupInformation->ConfigurationFile));
        }

        [Test]
        static void CanReadConfigFile()
        {
            Assert::That(ConfigurationManager::AppSettings->Get("test.setting"), Is::EqualTo("54321"));
        }

	};
}
